implicits/src-3/upickle/implicits/CaseClassReader.scala

```diff
@@ -82,11 +99,27 @@ trait CaseClassReaderPiece extends MacrosCommon:
       Reader.merge[T](readers:_*)
   }
 
+  inline def macroEnumR[T](using m: Mirror.Of[T]): Reader[T] = inline m match {
+    case m: Mirror.ProductOf[T] =>
+      throw new UnsupportedOperationException("Generated Enum Writer should never encounter Product")
+
+    case m: Mirror.SumOf[T] =>
+      val valueOf = macros.enumValueOf[T]
+      val description = macros.enumDescription[T]
+      new EnumReader[T](valueOf, description)
+  }
+
   inline given [T <: Singleton: Mirror.Of]: Reader[T] = macroR[T]
 
   // see comment in MacroImplicits as to why Dotty's extension methods aren't used here
   implicit class ReaderExtension(r: Reader.type):
-    inline def derived[T](using Mirror.Of[T]): Reader[T] = macroR[T]
+    inline def derived[T](using Mirror.Of[T]): Reader[T] = inline erasedValue[T] match {
+      case _: scala.reflect.Enum =>
```

Comment:
as mentioned in my comment above, for backwards-compatibility/upgradability, I think that the derivation logic should all be contained in the macroR method, and this should be a simple forwarder to it